docs/designs/dependency-repository.md

```diff
@@ -0,0 +1,100 @@
+# Dependency Repository
+
+Docfx support to build repo content which has `dependencies` to another repository. The `dependencies` relationships include
+
+  - Inclusion (token/code-snippet files in another repository)
+  - Content (articles/resource files in another repository)
+
+## How to config
+
+Define the dependent repositories in `docfx.yml` `dependencies` configuration with alias:
+
+Example:
+
+```yml
+dependencies:
+  _csharplang: "https://github.com/dotnet/csharplang#master"
+```
+Or
+
+```yml
+dependencies:
+  _csharplang: 
+    remote: https://github.com/dotnet/csharplang
```

Comment:
consider `url` or `remoteUrl` since `remote` could mean the name of the remote like `origin`, may have to update `PackageUrl` accordingly

Reply:
let's use remoteUrl